### PR TITLE
E2E Improvements and groundwork for test-pad tool

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
Add several features to enable upcoming `test-pad` tool.
- Added `rancher.sh` script, allowing rancher to be installed on a single node k3s server after startup
- Fixed split-server test, added proper node-taints that rancher also uses
- Added option to have "full" servers in the split-server config
- Added alpine support for rapid vm startup
- Reverted back to Ubuntu for the default OS, vagrant boxes from leap are not being updated anymore and are not stable with the latest release of vagrant

#### Types of Changes ####
E2E testing improvements

#### Verification ####
All E2E tests still pass

#### Linked Issues ####
[N/A](https://github.com/rancher/ecm-distro-tools/issues/89)

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/e2e/scripts/rancher.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+node_ip=$1
+
+curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
```
Comment (brandond) [inline tests/e2e/scripts/rancher.sh:7]: Maybe just drop a manifest?
```yaml
---
apiVersion: v1
kind: Namespace
metadata:
  name: cert-manager
---
apiVersion: v1
kind: Namespace
metadata:
  name: cattle-system
---
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  namespace: kube-system
  name: cert-manager
spec:
  targetNamespace: cert-manager
  version: v1.6.1
  chart: cert-manager
  repo: https://charts.jetstack.io
  set:
    installCRDs: "true"
---
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: rancher
  namespace: kube-system
spec:
  targetNamespace: cattle-system
  version: 2.6.5
  chart: rancher
  repo: https://releases.rancher.com/server-charts/latest
  set:
    ingress.tls.source: "rancher"
    hostname: "localhost"
    antiAffinity: "required"
    replicas: 1
```

**Review by briandowns — APPROVED** [review]

File: tests/e2e/scripts/rancher.sh (hunk 1 above)
Comment (dereknola) [reply tests/e2e/scripts/rancher.sh]: This is a much better idea, I have ripped out the helm install and just embedded a manifest into the rancher.sh script.

File: tests/e2e/scripts/rancher.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+node_ip=$1
+
+echo  "Give K3s time to startup"
+sleep 15
+
+cat << EOF > rancher.yaml
```
Comment (brandond) [inline tests/e2e/scripts/rancher.sh:7]: If this is running on the node itself, can we let the deploy controller pick it up?

```suggestion
cat << EOF > /var/lib/rancher/k3s/server/manifests/rancher.yaml
```

File: tests/e2e/scripts/rancher.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  version: 2.6.5
+  chart: rancher
+  repo: https://releases.rancher.com/server-charts/latest
+  set:
+    ingress.tls.source: "rancher"
+    hostname: "$node_ip.nip.io"
+    antiAffinity: "required"
+    replicas: 1
+EOF
+kubectl apply -f rancher.yaml
```
Comment (brandond) [inline tests/e2e/scripts/rancher.sh:48]:
```suggestion
```

**Review by brandond — APPROVED** [review]

**Pull request merged**